docs/ProviderDesign.md

```diff
@@ -0,0 +1,79 @@
+# Provider's view of the FLOCX
+
+![ProviderDesign](../images/ProviderService-Marketplace.png)
+
+
+
+
+## Multi-tenant IRONIC:
+Ironic manages multiple pools of hardware each owned by a separate user/group/organization.
+
+
+## Provider Service:
+A centralized service per OpenStack deployment that manages offers and contracts 
+on behalf of all its tenants of the OpenStack.
+
+## Keystone:
+Provides authentication and authorization services.
+
+
+## Workflow:
+
+Following steps describe the process of offering physical hardware in the market by individual owners
+and receiving contracts from the marketplace.
+
+
+
+1. Hardware pool owners decide which nodes to offer in the marketplace and submit offers to the provider service
+2. Provider service confirms owner-hardware relationship before accepting offers. 
+3. For each accepted offer it fetches the hardware configuration from Ironic
+4. These offers are published to the marketplace as soon as possible. 
+5. Receives contracts from the marketplace.
+
+
+## Offers (Owners --> Providers):
+An offer made by an owner and submitted to the provider service will have following attributes.
+* Node ID: Unique ID of the node
+* Start time: Date and time of availability
+* Duration: Hours, days that the node will be available
+* Floor Price: Minimum Price that is demanded.
+
+Provider validates the offer received from the owner  from keystone and adds the following attributes:
+* Offer ID: Unique ID for each offer.
+* Created_at:
+* Status: Waiting 
+
+    * Possible values: 
+    * Waiting: When provider accepts offer from the owner
+    * Published: When provider publishes the offer in the marketplace
+    * InContract: When an offer finds a suitable bid and a contract is formed.
+    * Expired: For any reason if the offer is cancelled or gets past its time of availability or gets used.
+
+Provider also fetches the node configuration from Ironic and adds it to the offer. 
```

Comment:
I don't think this is quite right; this information isn't added to the provider's offer. Rather, when the offer is submitted to the marketplace, the Ironic information is submitted as well.

Reply:
The context is what actions does the provider service take on receiving the offer from the owner.
it 
1. verifies if the owner is authorized to make an offer
2. extracts hw_config from ironic and _appends_ it to the offer
3. Publishes the offer to the marketplace.

Reply:
"appends it to the offer" implies that the provider offer has the ironic data added to it (at least it does to me). I'd argue that's not really what's happening. Instead, the provider uses the provider offer and the ironic information to create a marketplace offer.